lib/msal-react/src/useMsalAuthentication.tsx

```diff
@@ -0,0 +1,44 @@
+import { useCallback, useEffect } from "react";
+import { PopupRequest, RedirectRequest, SsoSilentRequest, InteractionType } from "@azure/msal-browser";
+import { AccountIdentifiers, useIsAuthenticated } from "./useIsAuthenticated";
+import { useMsal } from "./MsalProvider";
+import { IMsalContext } from "./MsalContext";
+
+type MsalAuthenticationResult = {
+    msal: IMsalContext;
+};
+
+export function useMsalAuthentication(
+    interactionType: InteractionType, 
+    authenticationRequest?: PopupRequest|RedirectRequest|SsoSilentRequest, 
+    accountIdentifier?: AccountIdentifiers
+): MsalAuthenticationResult {
+    const msal = useMsal();
+    const isAuthenticated = useIsAuthenticated(accountIdentifier);
+
+    const login = useCallback(() => {
+        switch (interactionType) {
+            case InteractionType.Popup:
+                return msal.instance.loginPopup(authenticationRequest as PopupRequest);
```

Comment:
Since `authenticationRequest` is an optional, does this work by supplying a default `PopupRequest` if none is given?

Reply:
The request object is also optional the login APIs, if not provided msal-browser will fill in things like `authority` and `redirectUri` from the values provided on the PublicClientApplication config and default the `scopes` to `openid` and `profile`